src/components/data-table/column-sorting/column-sort.directive.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Directive,
+  Input,
+  HostBinding,
+  HostListener,
+  OnDestroy
+} from "@angular/core";
+import {Subscription} from 'rxjs/Subscription';
+import {
+  SortDirection,
+  IColumnSortingModel
+} from './column-sort';
+import {MdDataColumnSortingService} from './column-sort.service';
+
+/**
+ * @name mdDataSortColumnDirective
+ *
+ * @description
+ * A directive that styles its host with sorting class names, and calls
+ * {@link MdDataColumnSortingService.changeSorting} to the service instance in its
+ * execution context.
+ *
+ * @usage
+ *
+ * <hljs lang="html">
+ *  <tr>
+ *    <th md-data-column-sort="1">Cheese</th>
```

Comment:
I had numerical IDs, but decided they should be strings to be more flexible. I can indicate that here as well.